pylib/gyp/common.py

```diff
@@ -422,17 +422,62 @@ def EnsureDirExists(path):
     except OSError:
         pass
 
+def GetCrossCompilerPredefines():
+    CC = os.environ.get("CC_target") or os.environ.get("CC")
+    CFLAGS = os.environ.get("CFLAGS")
+    CXX = os.environ.get("CXX_target") or os.environ.get("CXX")
+    CXXFLAGS = os.environ.get("CXXFLAGS")
+    cmd = []
+
+    if CC:
+        cmd += CC.split(" ")
+        if CFLAGS:
+            cmd += CFLAGS.split(" ")
+    elif CXX:
+        cmd += CXX.split(" ")
+        if CXXFLAGS:
+            cmd += CXXFLAGS.split(" ")
+    else:
+        return None
 
-def GetFlavor(params):
+    if sys.platform == "win32":
+        fd, input = tempfile.mkstemp(suffix=".c")
+        try:
+            os.close(fd)
+        except Exception:
```

Comment:
PEP8 warns about broad exceptions like this but we can let it slide...
* https://realpython.com/the-most-diabolical-python-antipattern